### ISSUE TITLE
Restart functionality, a la AwesomeWM

### ISSUE THREAD

**Issue by dunmatt** [sminez/penrose#70]
### Describe the change / addition you'd like to see made
It'd be great if it was possible to restart a Penrose WM without closing and reopening everything, the same as you can with AwesomeWM.  That way when you add some new hotkey you can simply run `cargo install --path .` followed by whatever hotkey is configured to trigger the restart action.

### Describe alternatives you've considered

Alternatively I can just keep doing it the way I have been; logging out and back in.

### Additional context

I have no idea if this is even possible to do, it could be that Awesome does it by cheating (not really restarting, but merely dropping state and re-reading the lua script) and that I'm asking the impossible.

One way or the other, keep up the great work, Penrose is a thing of beauty!  If there's a way I can help I'd be glad to have a go at it.

**Comment by sminez** [comment]
Hi @dunmatt sorry about not responding to this earlier, I've not been great at keeping up with GitHub notifications recently!

I've looked into a couple of ways of doing this previously and sadly it's not as simple as it first seems. I had a stab at recreating the behaviour used in `dwm` which is to query the X server for existing clients on startup and then handle what is found as if the clients had just been spawned, but that had a bunch that messed with layouts which I couldn't work out so I dropped it. To be honest, given that penrose is compiled to a binary (and this restart clunkyness is only an issue when hacking on said binary instead of day to day use) I'm tempted to mark this as "won't do" unless there is something additional about this feature that I'm missing :slightly_smiling_face:
